lld/MachO/Arch/X86_64.cpp

```diff
@@ -84,6 +84,8 @@ int64_t X86_64::getEmbeddedAddend(MemoryBufferRef mb, uint64_t offset,
   const uint8_t *loc = buf + offset + rel.r_address;
 
   switch (rel.r_length) {
+  case 0:
+    return *loc + pcrelOffset(rel.r_type);
   case 2:
     return static_cast<int32_t>(read32le(loc)) + pcrelOffset(rel.r_type);
```

Comment:
```suggestion
    return static_cast<int64_t>(read32le(loc)) + pcrelOffset(rel.r_type);
```

Each case adds `pcrelOffset(rel.r_type)`. Does it make sense to factor this out by adding a new address variable and `return address + pcrelOffset(rel.r_type);` at the end?

Reply:
the static cast here is correct. `read32le` returns an unsigned int32_t but addends can be negative, so we need to cast to int32_t to handle them correctly.

> Does it make sense to factor this out by adding a new address variable and return address + pcrelOffset(rel.r_type); at the end?

sure, done

Reply:
Oh that makes sense. In that case, can the single byte addend be negative? I believe it's read as an unsigned byte currently.